lib/github/middleware.js

```diff
@@ -60,13 +65,13 @@ module.exports = function middleware(callback) {
     const gitHubInstallationId = context.payload.installation.id;
     context.log = context.log.child({ gitHubInstallationId });
 
-    if (context.payload.sender.type === 'Bot') {
-      context.log({ noop: 'bot', botId: context.payload.sender.id, botLogin: context.payload.sender.login }, 'Halting futher execution since the sender is a bot');
+    if (context.payload.sender.type === 'Bot' && !isStateChangeAction(context.payload.action)) {
```

Comment:
@atrigueiro can we add a comment somewhere explaining why we're doing this? I.e. explain that the app can trigger pull request `edit` events which can cause bot loops and why it's safe to process bot events that are not in the list above.

Reply:
@thombergs I think you might be confused with the logic here. The opposite is actually happening. We won't allow edit events because they can cause bot loops and we're also not processing any other event - except the ones from that list -which are a one-time action therefore they won’t cause a loop.